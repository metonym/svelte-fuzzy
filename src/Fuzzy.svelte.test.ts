// tslint:disable-next-line: no-submodule-imports
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/svelte';
import SvelteFuzzy from './Fuzzy.svelte';

it('returns the correct result', () => {
  const data = [
    { title: "Old Man's War" },
    { title: 'The Lock Artist' },
    { title: 'HTML5' }
  ];
  const options = { keys: ['title'] };

  const { container, component, rerender } = render(SvelteFuzzy, {
    props: {
      query: 'old',
      data,
      options
    }
  });

  expect(component.$$.ctx.result).toEqual([
    {
      title: [
        { matches: true, text: 'Old' },
        { matches: false, text: " Man's War" }
      ]
    }
  ]);

  rerender({ props: { query: '' } });
  expect(container.firstChild).toHaveTextContent('');
});
