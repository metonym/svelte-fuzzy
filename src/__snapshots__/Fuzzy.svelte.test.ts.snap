// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query "old" 1`] = `"<div><div class=\\"svelte-fuzzy\\"><div class=\\"svelte-fuzzy-field\\"><span class=\\"svelte-fuzzy-title\\"><mark>Old</mark></span><span class=\\"svelte-fuzzy-title\\"> Man's War</span></div><div class=\\"svelte-fuzzy-field\\"><span class=\\"svelte-fuzzy-author\\">John</span></div> </div></div>"`;

exports[`query "remy" 1`] = `"<div><div class=\\"svelte-fuzzy\\"><div class=\\"svelte-fuzzy-field\\"><span class=\\"svelte-fuzzy-title\\"><mark>Remy</mark></span></div><div class=\\"svelte-fuzzy-field\\"><span class=\\"svelte-fuzzy-author\\"><mark>Remy</mark></span></div> </div></div>"`;
